### PR TITLE
feat: update configs to be able to support Deneb on all networks

### PR DESCRIPTION
**Pull request by nflaig**
**Motivation**

Currently it is not possible to support Deneb on other networks like Gnosis as we define values that need to be part of configuration as constants and hence those can't be customized.

**Description**

- Move `MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS` from constants to config, this value is defined as part of configuration in the spec ([specs/deneb/p2p-interface.md?plain=1#L66](https://github.com/ethereum/consensus-specs/blob/3727a757450feb6110e2d4e403604572a478343d/specs/deneb/p2p-interface.md?plain=1#L66))
- Align structure / comments with consensus spec
- Add references to preset / config files
- Rename directory and files that contain config values (presets --> configs), the consensus spec is pretty clear on the terminology ([presets](https://github.com/ethereum/consensus-specs/tree/dev/presets) vs. [configs](https://github.com/ethereum/consensus-specs/tree/dev/configs)) and this avoids potential confusion in the code

**Support all network configs**

After reviewing the spec values I noticed we are still missing plenty of network values which should be part of the config ([configs/mainnet.yaml#L113-L150](https://github.com/ethereum/consensus-specs/blob/3727a757450feb6110e2d4e403604572a478343d/configs/mainnet.yaml#L113-L150)). We either don't support them or those are defined as constants. 

I looked into refactoring this to move all of the constants into config but not all parts of the code that use those values can get access to configs easily, for example values used in `rateLimitQuotas`
https://github.com/ChainSafe/lodestar/blob/2db7120f3534faf5c023954f4065912dd0c2691f/packages/beacon-node/src/network/reqresp/rateLimit.ts#L11
While it should be possible to refactor the code to make this work, I think we should avoid large changes before the Deneb hard fork as current configuration is well tested already and there is no specific requirements to move those values from constants to config.

**BREAKING CHANGE**

Mainnet and minimal configs are now exported from `/configs` sub path instead of `/presets`.

The following change needs to be done if you are directly importing these from the `@lodestar/config` package 
```diff
-import {minimalChainConfig, mainnetChainConfig} from "@lodestar/config/presets";
+import {minimalChainConfig, mainnetChainConfig} from "@lodestar/config/configs";
```

### CODE REVIEW

File: packages/validator/src/util/params.ts [hunk 1]
```diff
@@ -121,14 +121,17 @@ function getSpecCriticalParams(localConfig: ChainConfig): Record<keyof ConfigWit
...
+    // Fork choice
     PROPOSER_SCORE_BOOST: false, // Ignored as it's changing https://github.com/ethereum/consensus-specs/pull/2895
 
     // Deposit contract
     DEPOSIT_CHAIN_ID: false, // Non-critical
     DEPOSIT_NETWORK_ID: false, // Non-critical
     DEPOSIT_CONTRACT_ADDRESS: true,
 
+    // Networking
+    MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS: false,
```
Comment (nflaig) [inline packages/validator/src/util/params.ts:133]: @g11tech Based on my understanding of this value it is not critical to verify it here as it is only relevant for the beacon node and as part of networking can't result in a consensus failure.

Maybe you could confirm this, and should we add a comment here?

**Review by wemeetagain — COMMENTED** [review]
LGTM modulo @g11tech 's approval / comment

File: packages/validator/src/util/params.ts (hunk 1 above)
Comment (g11tech) [reply packages/validator/src/util/params.ts:133]: yes it doesn't affect consensus, adding with `false` is ok

Comment (nflaig) [reply packages/validator/src/util/params.ts:133]: I added a comment in https://github.com/ChainSafe/lodestar/pull/6328/commits/e94355bd2189a3e035899fe0f4c4530a126eeabd

**Review by wemeetagain — APPROVED** [review]

**Review by g11tech — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.15.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.15.0) :tada:

**Pull request merged**